File: android/app/src/main/kotlin/com/voice2text/app/contracts/AudioContract.kt
package com.voice2text.app.contracts

object AudioContract {
    const val RECORDER_CHANNEL = "voice2text/recorder"

    const val RECORDING_DIR_NAME = "recordings"
    const val RECORDING_EXTENSION = "m4a"

    const val SAMPLE_RATE_HZ = 16000
    const val BIT_RATE = 64000
    const val CHANNEL_COUNT = 1

    const val CONTAINER_FORMAT = "mpeg4"
    const val CODEC = "aac"
}

File: lib/app/contracts/audio_contract.dart
class AudioContract {
  static const String recorderChannel = 'voice2text/recorder';

  static const String recordingDirName = 'recordings';
  static const String recordingExtension = 'm4a';

  static const int sampleRateHz = 16000;
  static const int bitRate = 64000;
  static const int channelCount = 1;

  static const String containerFormat = 'mpeg4';
  static const String codec = 'aac';

  // Keep this up-to-date with Android native implementation.
  static const String compatibilityNote =
      'Current recorder output is m4a/aac @16kHz mono for Sherpa bridge stub.';
}
